### DISCUSSION TITLE
sunxi/nanopineo2 gpio

### DISCUSSION THREAD

**Discussion by vector-co-uz** [immortalwrt/immortalwrt#699] [Q&A]
I have nanopi neo2 with neo [dock](https://wiki.friendlyelec.com/wiki/index.php/1-bay_NAS_Dock_v1.2_for_NanoPi_NEO/NEO2). I want to connect the ssd drive to the dock, but openwrt does not detect sata and usb socket.

I was told that GPIO is not defined at all in the system. and I was advised to rebuild with GPIO support.
**My question is: does the kernel itself have support?
Are there required drivers? (Controller on dock is JMicron ). And what do I need to focus on to enable support?**

**Comment by 1715173329** [comment]
I just found you have created a similar issue on upstream openwrt: openwrt/openwrt#9961.

I don't know what it is, I have neither the device nor dock. I'd like to suggest you contact with the manufactory for help.

**Reply by vector-co-uz** [reply]
I already asked my question there. So far there is no answer.

**Comment by vector-co-uz** [comment]
Can someone tell me, for Pine64 plus, Orange PC2 or nanopi r1s, how are gpio ports determined?  which module is responsible for this?  They are on the same target, maybe this will help me?
